src/init.cpp

```diff
@@ -2000,6 +2000,13 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
         if (connect.size() != 1 || connect[0] != "0") {
             connOptions.m_specified_outgoing = connect;
         }
+        if (!connOptions.m_specified_outgoing.empty() && !connOptions.vSeedNodes.empty()) {
+            LogPrintf("-seednode is ignored in -connect mode\n");
+        }
+
+        if (args.GetBoolArg("-dnsseed", DEFAULT_DNSSEED) && args.IsArgSet("-proxy")) {
```

Comment:
But this is true even if no -connect is configured? Then the warning is confusing...
Perhaps move this into the block above (same suggestion there)?

Also, if `DEFAULT_DNSSEED` is ever set to `true`, this will become confusing I guess every time proxy is configured, but i'm not sure it's worth addressing. Ideally, we should just read `-dnsseed` here without looking at the default value.

Reply:
Warning said `-dnsseed` is ignored. But it will be shown even if a user haven't touched `-dnsseed` at all, just when they configured a `-proxy`. Feels like bad UX.

Reply:
Reg. the first comment: If the configuration is `-dnsseed=1 -proxy=X`, `ThreadDNSAddressSeed` will queue `ADDR_FETCH` connections and they will get processed because `ThreadOpenConnections` will invoke `ProcessAddrFetch` with some frequency. (This will no longer happen when using `-connect=X -dnsseed=1 -proxy=X` after this PR is merged)

Reg. the second comment: You're right. If the configuration is `-connect=X -proxy=Y`, the log would print and the user might be confused. I have updated the code to check explicitly that the user set the arg. I have also added a test case to confirm the same.